x-pack/plugin/inference/qa/inference-service-tests/src/javaRestTest/java/org/elasticsearch/xpack/inference/InferenceBaseRestTest.java

```diff
@@ -72,15 +72,23 @@ protected Map<String, Object> putModel(String modelId, String modelConfig, TaskT
         var request = new Request("PUT", endpoint);
         request.setJsonEntity(modelConfig);
         var reponse = client().performRequest(request);
-        assertOkWithErrorMessage(reponse);
+        assertOkOrCreated(reponse);
```

Comment:
> BTW, why isn't this called just parseConfig?

There is a common pattern we use where we have different parsers for the PUT request and for reading the persisted config back. This is because the validation is stricter on PUT - we only accept known fields. However in a rolling upgrade scenario in a mixed cluster an new node may have PUT the config with extra fields which are not known to the old node. So when the old node reads persisted config that has been created by a new node we tolerate unknown fields.